src/main/java/com/hubspot/jinjava/objects/collections/SizeLimitingPyList.java

```diff
@@ -31,25 +35,31 @@ public SizeLimitingPyList(List<Object> list, int maxSize) {
   }
 
   @Override
-  public boolean add(Object element) {
+  public boolean add(@Nonnull Object element) {
     checkSize(size() + 1);
     return super.add(element);
   }
 
   @Override
-  public void add(int index, Object element) {
+  public void add(int index, @Nonnull Object element) {
```

Comment:
Why would you not be able to add a `null` object to a list?
I think this would break some usages because we currently allow null objects in lists.

Reply:
Good point, though I don't think the annotation will break anything, especially when called from jinjava templates.

Reply:
Actually, I don't think this annotation actually does anything during runtime so it wouldn't break anything, but my point remains about allowing `null` elements

Reply:
Actually, I added this only because of the intellij warning `Not annotated parameter overrides @ParametersAreNonnullByDefault parameter`

Reply:
I don't see that annotation, but I see ForwardingList annotation [@ParametricNullness](https://cs.android.com/android/platform/superproject/+/master:external/guava/android/guava/src/com/google/common/hash/ParametricNullness.java):
> Marks a "top-level" type-variable usage as the closest we can get to "non-nullable when non-nullable; nullable when nullable"

I don't understand what that means

Reply:
Seems okay either way, really